### ISSUE TITLE
fromJS() only works on Objects in the same Realm

### ISSUE THREAD

**Issue by gajus** [immutable-js/immutable-js#1039]
### How to reproduce

```js
const Immutable = require('immutable');
const vm = require('vm');

console.log('OUT', Immutable.Iterable.isIterable(Immutable.fromJS({})) );

vm.runInNewContext(`
  console.log('IN', Immutable.Iterable.isIterable(Immutable.fromJS({})) );
`, {
  console,
  Immutable
}, {});

```

### Output

```
OUT true
IN false
```

### Expected output

```
OUT true
IN true
```

**Comment by gajus** [comment]
It all comes down to `value.constructor === Object` in [/src/fromJS.js#L39](https://github.com/facebook/immutable-js/blob/29a5c55727c523b0d83c53c7ccd1032aaa579f49/src/fromJS.js#L39). Outside of VM it executes to `true` and inside the VM it executes to `false`.

I have tried adding `Object` to the sandbox, but it didn't change the outcome.

**Comment by gajus** [comment]
Even this doesn't work:

```js
const vm = require('vm');

vm.runInNewContext(`
  const Immutable = require('immutable');

  console.log( Immutable.Iterable.isIterable(Immutable.fromJS({})) );
`, {
  console,
  require,
});

```

**Comment by gajus** [comment]
Interesting to note that the following evaluates to `true`:

```js
const vm = require('vm');

vm.runInNewContext(`
  console.log( test.constructor === Object );

`, {
  console,
  test: {},
  Object
});
```

**Comment by gajus** [comment]
Oh, I see:

```
const vm = require('vm');

vm.runInNewContext(`
  console.log( {}.constructor === Object );
`, {
  console,
  Object
});

```

This evaluates to `false`. Thats explain why. Not sure how to fix it, though.

**Comment by gajus** [comment]
@leebyron Is there any interest in getting this fixed?

**Comment by leebyron** [comment]
For sure. #1056 is one attempt, however it has some punishing performance qualities.

I'm open to more exploration on this topic

**Comment by turnerhayes** [comment]
I suspect this is what's causing my custom REPL (using the [REPL](https://nodejs.org/api/repl.html) module) to exhibit the same issue--presumably, the REPL module is using a VM under the hood. Using the regular Node REPL (i.e. just invoking `node` on the command line) doesn't have this issue.

**Comment by turnerhayes** [comment]
Although when I run `({x: 1}).constructor === Object` from within my custom REPL, it returns `true`, which sounds contrary to the diagnosis here, but maybe there's some weirdness around closures or something.

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/954d75b02c6a86c6694a1409594c42f406640100

**Comment by jdeniau** [comment]
Hi!
The 4.0.0-rc.14 version has been released and should fix this issue.
Feel free to test it and re-open this issue if the issue is still present.
Thank you!
